pkg/sloop/storemanager/storemanager.go

```diff
@@ -193,92 +232,105 @@ func doCleanup(tables typed.Tables, timeLimit time.Duration, sizeLimitBytes int,
+	}
 
-		glog.Infof("Remaining number of keys to delete: %v  ", numOfKeysToDeleteForFileSizeCondition)
+	// if all the partitions have to be cleaned uo there is no need to further check for time condition
+	if len(partitionsToDelete) == len(sortedPartitionsList) {
+		return partitionsToDelete, partitionMap
 	}
 
-	elapsed := time.Since(beforeGCTime)
-	glog.Infof("Deletion of prefixes took %v and removed %d keys with error: %v", elapsed, totalNumOfDeletedKeys, err)
+	// Is clean up condition still not met for partitions collected for size
+	currentLastPartitionToDeleteIndex := len(partitionsToDelete)
+	for currentLastPartitionToDeleteIndex < len(sortedPartitionsList) && cleanUpTimeCondition(sortedPartitionsList[currentLastPartitionToDeleteIndex], maxPartition, timeLimit) {
+		partitionsToDelete = append(partitionsToDelete, sortedPartitionsList[currentLastPartitionToDeleteIndex])
+		currentLastPartitionToDeleteIndex++
+	}
 
-	beforeDropPrefix := time.Now()
+	minPartitionAge, err := untyped.GetAgeOfPartitionInHours(sortedPartitionsList[currentLastPartitionToDeleteIndex])
+	if err == nil {
+		metricAgeOfMinimumPartition.Set(minPartitionAge)
+	}
 
-	// dropping prefix to force compression
-	err = tables.Db().DropPrefix([]byte{})
-	glog.Infof("Drop prefix took %v with error: %v", time.Since(beforeDropPrefix), err)
-	return anyCleanupPerformed, totalNumOfDeletedKeys, totalNumOfKeysToDelete, nil
+	return partitionsToDelete, partitionMap
 }
 
-func deletePartition(minPartition string, tables typed.Tables, deletionBatchSize int) (int, int, []string) {
-	totalNumOfDeletedKeysforPrefix := 0
-	totalNumOfKeysToDeleteForPrefix := 0
-	numOfDeletedKeysforPrefix := 0
-	numOfKeysToDeleteForPrefix := 0
+func deletePartition(minPartition string, tables typed.Tables, deletionBatchSize int, enableDeleteKeys bool, partitionInfo *common.PartitionInfo) (uint64, uint64, []string) {
+	var totalNumOfDeletedKeysForPrefix uint64 = 0
+	var totalNumOfKeysToDeleteForPrefix uint64 = 0
+	var numOfDeletedKeysForPrefix uint64 = 0
+	var numOfKeysToDeleteForPrefix uint64 = 0
 
 	partStart, partEnd, err := untyped.GetTimeRangeForPartition(minPartition)
 	glog.Infof("GC removing partition %q with data from %v to %v (err %v)", minPartition, partStart, partEnd, err)
 	var errMessages []string
 	for _, tableName := range tables.GetTableNames() {
 		prefix := fmt.Sprintf("/%s/%s", tableName, minPartition)
 		start := time.Now()
-		err, numOfDeletedKeysforPrefix, numOfKeysToDeleteForPrefix = common.DeleteKeysWithPrefix([]byte(prefix), tables.Db(), deletionBatchSize)
-		metricGcDeletedNumberOfKeysByTable.WithLabelValues(fmt.Sprintf("%v", tableName)).Set(float64(numOfDeletedKeysforPrefix))
+		numberOfKeysToRemove := partitionInfo.TableNameToKeyCountMap[tableName]
+		if enableDeleteKeys {
+			err, numOfDeletedKeysForPrefix, numOfKeysToDeleteForPrefix = common.DeleteKeysWithPrefix(prefix, tables.Db(), deletionBatchSize, numberOfKeysToRemove)
```

Comment:
when err != nil, we might want to log the error, and print out numOfDeletedKeysForPrefix, numOfKeysToDeleteForPrefix

Reply:
its already logged in DeleteKeysWithPrefix()

Reply:
No, it  did not, I see return err, where is the log part?